tests/system/common.py

```diff
@@ -771,6 +778,99 @@ def running_task_status(task_statuses):
     assert False, "Did not find a TASK_RUNNING status in task statuses: %s" % (task_statuses,)
 
 
+def unreachable_tasks(framework_name='marathon'):
+    """ Returns the unreachable tasks for a given framework.
+    """
+    client = mesos.DCOSClient()
+    frameworks = client.get_master_state()['frameworks']
+    for framework in frameworks:
+        if framework["name"] == framework_name:
+            return framework['unreachable_tasks']
+    return []
+
+def mesos_tasks(framework_name='marathon'):
+    client = mesos.DCOSClient()
+    frameworks = client.get_master_state()['frameworks']
+    for framework in frameworks:
+        if framework["name"] == framework_name:
+            return framework['tasks']
+    return []
+
+
+def unreachable_task_predicate(service):
+    return len(unreachable_tasks(service)) > 0
+
+
+def elapse_time(start, end, precision=0):
+    return round(end - start, precision)
+
+
+def wait_for_unreachable_task(service='marathon', timeout_sec=10*60, inverse=False):
+
+    report_freq = 5 if timeout_sec < 600 else 15
+    print("Starting wait for unreachable tasks")
+    @retrying.retry(wait_fixed=1000, stop_max_delay=timeout_sec*1000)
+    def __wait_for_unreachable_task(start=time.time(), msg='unreachable tasks'):
+        elapsed = elapse_time(start, time.time())
+        if (elapsed % report_freq == 0):
```

Comment:
Hm, this is a little confusing. What is the reason for not reporting this with each try?